scipy/interpolate/interpolate.py

```diff
@@ -36,13 +37,21 @@ def lagrange(x, w):
     ----------
     x : array_like
         `x` represents the x-coordinates of a set of datapoints.
-    w : array_like
-        `w` represents the y-coordinates of a set of datapoints, i.e., f(`x`).
+    y : array_like
+        `y` represents the y-coordinates of a set of datapoints, i.e., f(`x`).
+    reduce : bool
+        `reduce = True` represents use reduced Lagrange coefficients.
+        `reduce = False` represents use complete Lagrange coefficients (the size
+        of the coefficient array equals the size of x).
+        Default is `True`.
 
     Returns
     -------
     lagrange : `numpy.poly1d` instance
         The Lagrange interpolating polynomial.
+    coeffs : ndarray
+        If `reduce = True`, only `lagrange` is returned.
+        Otherwise, a tuple `(lagrange, coeffs)` is returned.
```

Comment:
Yuck, variable-length returns. Makes me question the whole premise for the patch TBH.

Reply:
> Yuck, variable-length returns. Makes me question the whole premise for the patch TBH.

It seems difficult to ensure the consistent variable length to return (if inheritance from `poly1d` is not used) because the return value `p` in the original code is an object of  the class `poly1d` and for back-compatibility reasons.

Reply:
@ev-br I try to ensure the consistent length of the return value, but I feel it thorny. Do you have any good way or suggestion?

Reply:
Nope, as long as it returns a poly1d object, which is frozen by backcompat